src/servers/graphql-old-server.ts

```diff
@@ -297,33 +310,41 @@ const resolvers = {
         if (result instanceof Error) throw result
         return result
       },
-      payInvoice: async ({ invoice, amount, memo }) => {
-        const decodedInvoice = await decodeInvoice(invoice)
-        if (decodedInvoice instanceof Error) throw decodedInvoice
-
-        const { amount: lnInvoiceAmount } = decodedInvoice
-        if (lnInvoiceAmount && lnInvoiceAmount > 0) {
-          const status = await lnInvoicePaymentSend({
-            paymentRequest: invoice,
-            memo,
-            walletId: wallet.user.id as WalletId,
-            userId: wallet.user.id as UserId,
-            logger,
-          })
-          if (status instanceof Error) throw mapError(status)
-          return status.value
-        }
-        const status = await lnNoAmountInvoicePaymentSend({
-          paymentRequest: invoice,
-          memo,
-          amount,
-          walletId: wallet.user.id as WalletId,
-          userId: wallet.user.id as UserId,
-          logger,
-        })
-        if (status instanceof Error) throw mapError(status)
-        return status.value
-      },
+      payInvoice: async ({ invoice, amount, memo }, _, { ip, domainUser }) =>
+        addAttributesToCurrentSpanAndPropagate(
+          {
+            [SemanticAttributes.ENDUSER_ID]: domainUser?.id,
+            [ENDUSER_ALIAS]: domainUser?.username,
+            [SemanticAttributes.HTTP_CLIENT_IP]: ip,
+          },
+          async () => {
+            const decodedInvoice = decodeInvoice(invoice)
+            if (decodedInvoice instanceof Error) throw decodedInvoice
+
+            const { amount: lnInvoiceAmount } = decodedInvoice
+            if (lnInvoiceAmount && lnInvoiceAmount > 0) {
+              const status = await lnInvoicePaymentSend({
+                paymentRequest: invoice,
+                memo,
+                walletId: wallet.user.id as WalletId,
+                userId: wallet.user.id as UserId,
+                logger,
+              })
+              if (status instanceof Error) throw mapError(status)
+              return status.value
+            }
+            const status = await lnNoAmountInvoicePaymentSend({
```

Comment:
Its being called here @nicolasburtey - that was the giant PR vindard was working on for a month. It was merged last week.

Reply:
thanks. I thought it would have been used only with api v2. 

we need to get rid of `core/` after v2 is been merged.